packages/connect/src/base/entry.ts

```diff
@@ -277,29 +514,59 @@ export class EntryNodes extends EventEmitter implements Initializable, Startable
    */
   protected async onNewRelay(peer: PeerStoreType): Promise<void> {
     if (peer.id.equals(this.getComponents().getPeerId())) {
+      // Cannot use self as entry node
       return
     }
+
     if (peer.multiaddrs == undefined || peer.multiaddrs.length == 0) {
-      log(`Received entry node ${peer.id.toString()} without any multiaddr`)
+      log(`Received entry node ${peer.id.toString()} without any multiaddr, ignoring`)
+      // Nothing to do
       return
     }
 
-    for (const uncheckedNode of this.uncheckedEntryNodes) {
-      if (uncheckedNode.id.equals(peer.id)) {
-        log(`Received duplicate entry node ${peer.id.toString()}`)
-        // TODO add difference to previous multiaddrs
-        return
-      }
-    }
+    let receivedNewAddrs = false
+
+    let knownPeer = false
+
+    this.someNode(
+      peer.id,
+      (entry: PeerStoreType | EntryNodeData) => {
+        knownPeer = true
+
+        log(
+          `Received new address${peer.multiaddrs.length == 1 ? '' : 'es'} ${peer.multiaddrs
+            .map((ma) => ma.decapsulateCode(CODE_P2P).toString())
+            .join(', ')} for ${peer.id.toString()}`
+        )
 
-    this.uncheckedEntryNodes.push({
-      id: peer.id,
-      multiaddrs: peer.multiaddrs.filter(isUsableRelay)
-    })
+        const existing = new Set(entry.multiaddrs.map((ma: Multiaddr) => ma.decapsulateCode(CODE_P2P).toString()))
```

Comment:
Could use a comment on what's happening on these lines and below.

Reply:
changed

Reply:
> Would be good to have a CHANGELOG entry. Please also add some documentation so it does not stay on this PR conversation only.
> 
> Before merging to `release/paleochora`, lets merge this to `master` (see #4011)

Added a CHANGELOG entry